src/utils/ExtrusionLine.cpp

```diff
@@ -40,19 +44,15 @@ coord_t ExtrusionLine::getMinimalWidth() const
                             })->w;
 }
 
-void ExtrusionLine::appendJunctionsTo(LineJunctions& result) const
-{
-    result.insert(result.end(), junctions.begin(), junctions.end());
-}
-
-
 void ExtrusionLine::simplify(const coord_t smallest_line_segment_squared, const coord_t allowed_error_distance_squared, const coord_t maximum_extrusion_area_deviation)
 {
-    if (junctions.size() <= 3)
+    if (junctions.size() <= 2 + is_closed)
     {
         return;
     }
 
+    // TODO: allow for the first point to be removed in case of simplifying closed Extrusionlines.
```

Comment:
It's still a TODO. I didn't want to burn my fingers on changing any simplify code, because historically that function has been very bug-prone!

I think there is code duplication between ExtrusionLine::simplify and Polygons::simplify. Perhaps we should merge those two into one templated function. However, I didn't want to add al that complexity to this PR.

What does the Cura team think about this? Should I fix the simplify function in this PR, or would that be out of scope?

Reply:
It should be noted that before this PR the wall lines would be connected in the PathOrderOptimizer, so the simplification never happened across ends of polylines. In that regard I didn't introduce this bug, but only reduced its impact, seeing as there are now less ends to the extrusionLines because they are stitched.